parla/task_runtime.py

```diff
@@ -428,84 +422,84 @@ def is_assigned(self):
-            return self._add_successor(successor)
+            return self._add_dependent(dependent)
 
-    def _add_successor(self, successor: "Task") -> bool:
-        """Add the successor if self is not completed, otherwise return False."""
+    def _add_dependent(self, dependent: "Task") -> bool:
+        """Add the dependent if self is not completed, otherwise return False."""
         if self._state.is_terminal:
             return False
-        logger.debug("Task %s added a successor %s", self.name, successor.name)
-        self._successors.append(successor)
+        logger.debug("Task %s added a dependent %s", self.name, dependent.name)
+        self._dependents.append(dependent)
         return True
 
-    def _notify_successors_mutex(self, events=None):
+    def _notify_dependents_mutex(self, events=None):
         with self._mutex:
-            self._notify_successors(events)
+            self._notify_dependents(events)
 
-    def _notify_successors(self, events=None):
-        for successor in self._successors:
-            successor._handle_predecessor_event_mutex(events)
-        self._successors = []
+    def _notify_dependents(self, events=None):
+        for dependent in self._dependents:
+            dependent._handle_dependency_event_mutex(events)
+        self._dependents = []
 
-    def _add_predecessor_mutex(self, dependency) -> bool:
+    def _add_dependency_mutex(self, dependency) -> bool:
         with self._mutex:
-            return self._add_predecessor(dependency)
+            return self._add_dependency(dependency)
 
-    def _add_predecessor(self, predecessor: 'Task', symmetric=True) -> bool:
+    def _add_dependency(self, dependency: 'Task', symmetric=True) -> bool:
         """
-        :param symmetric: If true, add self as succesor to predecessor as well
+        :param symmetric: If true, add self as succesor to dependency as well
         """
-        self._num_blocking_predecessors += 1
-        self._predecessors.append(predecessor)
-        if symmetric and not predecessor._add_successor_mutex(self):
-            self._num_blocking_predecessors -= 1
+        self._num_blocking_dependencies += 1
+        self._dependencies.append(dependency)
+        if symmetric and not dependency._add_dependent_mutex(self):
+            self._num_blocking_dependencies -= 1
             return False
         return True
 
-    def _handle_predecessor_event_mutex(self, events):
+    def _handle_dependency_event_mutex(self, events):
         with self._mutex:
-            self._num_blocking_predecessors -= 1
+            self._num_blocking_dependencies -= 1
             # Add events from one dependent task.
             # (We are aiming to multiple device tasks, and it would
             #  be possible to have multiple events)
             if events is not None:
-                self._predecessor_events.append(events)
+                self._dependency_events.append(events)
             self._maybe_ready_to_schedule()
```

Comment:
There was a comment on the old PR about this needed a better name, just putting it here again.